### PR TITLE
[rag eval] add overall questions feedback section

### PR DESCRIPTION
**Pull request by chejennifer**
add another eval stage for users to evaluate the questions asked by the LLM

Screenshot 2024-06-24 at 5 33 32 PM

### CODE REVIEW

File: static/js/apps/eval_retrieval_generation/overall_feedback.tsx [hunk 1]
```diff
@@ -21,44 +21,85 @@ import { Button } from "reactstrap";
...
 };
+// Dictionary of feedback stage -> title
+const QUESTION_TITLE = {
+  [FeedbackStage.OVERALL_ANS]: "OVERALL EVALUATION",
+  [FeedbackStage.OVERALL_QUESTIONS]: "QUESTIONS EVALUATION",
+};
+// Dictionary of feedback stage -> question
+const QUESTION = {
+  [FeedbackStage.OVERALL_ANS]: "How is the overall answer?",
+  [FeedbackStage.OVERALL_QUESTIONS]: "Are there any questions missing?",
```
Comment (pradh) [inline static/js/apps/eval_retrieval_generation/overall_feedback.tsx:72]: "Do the generated questions seem sufficient to answer the query?"

File: static/js/apps/eval_retrieval_generation/overall_feedback.tsx [hunk 2]
```diff
@@ -21,44 +21,85 @@ import { Button } from "reactstrap";
...
   },
 };
+// Dictionary of feedback stage -> title
+const QUESTION_TITLE = {
+  [FeedbackStage.OVERALL_ANS]: "OVERALL EVALUATION",
+  [FeedbackStage.OVERALL_QUESTIONS]: "QUESTIONS EVALUATION",
+};
+// Dictionary of feedback stage -> question
+const QUESTION = {
+  [FeedbackStage.OVERALL_ANS]: "How is the overall answer?",
```
Comment (pradh) [inline static/js/apps/eval_retrieval_generation/overall_feedback.tsx:71]: For RAG:

"How relevant is the overall answer to the query?"

**Review by pradh — APPROVED** [review]
Nice!

File: static/js/apps/eval_retrieval_generation/constants.tsx [hunk 3]
```diff
@@ -68,3 +70,7 @@ export const QUERY_OVERALL_OPTION_IRRELEVANT = "LLM_ANSWER_IRRELEVANT";
 export const QUERY_OVERALL_OPTION_SOMEWHAT_RELEVANT =
   "LLM_ANSWER_SOMEWHAT_RELEVANT";
 export const QUERY_OVERALL_OPTION_RELEVANT = "LLM_ANSWER_RELEVANT";
+// Options for the query calls overall feedback
+export const OVERALL_QUESTIONS_OPTION_HAS_MISSING = "LLM_QUESTIONS_MISSING";
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/constants.tsx:74]: the constants starts to feel overwhelming, later on may think about simplifying them. Imagine if half of these need change, what would the change look like (or minimized).

**Review by shifucun — APPROVED** [review]

File: static/js/apps/eval_retrieval_generation/overall_feedback.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/overall_feedback.tsx]: done

File: static/js/apps/eval_retrieval_generation/constants.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/constants.tsx:76]: added a TODO to clean up the constants

**Pull request merged**